compiler/resolution/loopDetails.cpp

```diff
@@ -146,6 +146,49 @@ Symbol* collapseIndexVarReferences(Symbol* index)
   return index;
 }
 
+/* Return whatever is producing the Symbol iterator.
```

Comment:
How is this function specific to iterators?

Reply:
If it is not specific to iterators, it would be nice to have it as a globally-visible function, perhaps in symbol.h.

Reply:
Most of the work is actually in Symbol::getInitialization. But the way it ignores FLAG_TEMP variables is specific to what we actually need here and I'm not so confident it's broadly useful. We can certainly move it to symbol if we find more cases needing the same pattern, though.